src/connectionController.ts

```diff
@@ -1,23 +1,33 @@
 import * as vscode from 'vscode';
-import { v4 as uuidv4 } from 'uuid';
-import Connection from 'mongodb-connection-model/lib/model';
-import DataService from 'mongodb-data-service';
+import {
+  connect,
+  convertConnectionModelToInfo,
+  convertConnectionInfoToModel,
+  ConnectionInfo,
+  ConnectionOptions,
+  DataService,
+  getConnectionTitle
+} from 'mongodb-data-service';
+import ConnectionModel from 'mongodb-connection-model';
+import ConnectionString from 'mongodb-connection-string-url';
 import { EventEmitter } from 'events';
-import { promisify } from 'util';
+import { v4 as uuidv4 } from 'uuid';
 
 import { CONNECTION_STATUS } from './views/webview-app/extension-app-message-constants';
-import { ConnectionModel } from './types/connectionModelType';
 import { createLogger } from './logging';
-import { DataServiceType } from './types/dataServiceType';
 import { ext } from './extensionConstants';
-import { SavedConnection, StorageScope } from './storage/storageController';
-import SSH_TUNNEL_TYPES from './views/webview-app/connection-model/constants/ssh-tunnel-types';
-import { StatusView } from './views';
+import RawConnectionModel from './views/webview-app/connection-model/connection-model';
+import { StorageLocation, ConnectionsFromStorage } from './storage/storageController';
 import { StorageController, StorageVariables } from './storage';
+import { StatusView } from './views';
 import TelemetryService from './telemetry/telemetryService';
 
-const { name, version } = require('../package.json');
+// TODO: export these helpers from mongodb-data-service.
```

Comment:
Definitely in the scope of the unified connections epic, but probably as separate pr.

Reply:
Is there a ticket for this?

Reply:
Also, I opened https://github.com/mongodb-js/compass/pull/2647 for the relevant changes in Compass, maybe we can also already address them here (after that Compass PR is merged)? Then this becomes less urgent

Reply:
Here is the ticket: https://jira.mongodb.org/browse/COMPASS-5378

Reply:
And yes we can bump a new data service version here when https://github.com/mongodb-js/compass/pull/2647 is merged.

Reply:
No, I meant, since we’re deciding to duplicate the code here instead of exposing it in data-service right now, could we do those changes also to the duplicated code?

Reply:
Ahh got it! let's see maybe I will do COMPASS-5378 tomorrow, so we just remove the duplicate completely, otherwise yeah I can add your change here as well.